source/agora/common/TransactionManager.d

```diff
@@ -0,0 +1,211 @@
+
+    public void saveTransaction (Hash hash, Transaction tx)
+    {
+        this.txs[hash] = tx;
+    }
+
+    /*******************************************************************************
+
+        Return hash of transaction
+
+        Params:
+            tx = Type of `Transaction`
+
+        Return:
+            `Hash` of `Transaction`
+
+    *******************************************************************************/
+
+    public Hash getTxHash (Transaction tx)
+    {
+        return hashFull(tx);
+    }
+
+    /*******************************************************************************
+
+        Checks whether the transaction is coinbase
+
+        Params:
+            tx = Type of `Transaction`
+
+        Return:
+            Return true if this transaction is a coinbase
+
+    *******************************************************************************/
+
+    public bool isCoinbaseTx (Transaction tx)
+    {
+        return tx.inputs.length == 1 && tx.inputs[0] == Input.init;
+    }
+
+    /*******************************************************************************
+
+        Creates a new coinbase transaction
+
+        Params:
+            address = The public key that can redeem this output
+            value = The initial value
+
+        Return:
+            Return coinbase transation
+
+    *******************************************************************************/
+
+    public Transaction newCoinbaseTX (PublicKey address, Amount value = 0)
+    {
+        Transaction tx;
+        tx.inputs ~= Input(Hash.init, 0);
+        tx.outputs ~= Output(value, address);
+        return tx;
+    }
+
+    /*******************************************************************************
+
+        Find a output of transaction
+
+        Params:
+            hash = `Hash` of `Transaction`
+            index = Index of the `Output` in the `Transaction`
+
+        Return:
+            `Output` in the `Transaction`
+
+    *******************************************************************************/
+
+    public Output* findTxOutput (Hash hash, uint index)
+    {
+        if (auto tx = hash in this.txs)
+        {
+            if (index < tx.outputs.length)
+            {
+                return &tx.outputs[index];
+            }
+            else
+            {
+                return null;
+            }
+        }
+        else
+        {
+            return null;
+        }
+    }
+
+
+    /*******************************************************************************
+
+        Get sum of `Input`
+
+        Params:
+            tx = `Transaction`
+
+        Return:
+            Sum of `Input` in the `Transaction`
+
+    *******************************************************************************/
+    public long getInputSum (Transaction tx)
+    {
+        return
+            tx.inputs
+                .map!(a => findTxOutput(a.previous, a.index))
+                .filter!(a => a != null)
```

Comment:
```suggestion
                .filter!(a => a !is null)
```

@Geod24: wasn't there some warning / deprecation in the past about using `!=` with null?